lib/sanity/attributable.rb

```diff
@@ -22,33 +29,46 @@ def attributes
         @attributes ||= []
       end
 
-      def default_attributes
-        @defaults ||= {}
+      def aliased_attributes
+        @aliased_attributes ||= {}
       end
 
       private
 
-      def attribute(name, default: nil)
-        attributes << name
-        default_attributes.merge!("#{name}": default)
+      def attribute(name, default: nil, as: nil, type: nil)
+        attributes.push(name)
+
+        self.define_method("#{name}=") do |value|
+          @attributes[name] =
+            case type
+            when :boolean
+              value.to_s == 'true'
+            when :float
+              value.to_f
+            when :integer
+              value.to_i
+            when :time
+              Time.parse(value.to_s)
+            else
+              value
+            end
+        end
```

Comment:
I like the idea here 👍

Reply:
I just had a thought when using this, do you think we should have a type for `:string`. It's currently the default, but I figured it might be wierd that we don't have it here for consistency. Also I should probably update the documentation if we do that.